ocaml/sdk-gen/csharp/autogen/src/JsonRpc.cs

```diff
@@ -206,13 +206,13 @@ private T Rpc<T>(string callName, JToken parameters, JsonSerializer serializer)
             webRequest.Proxy = WebProxy;
             webRequest.KeepAlive = KeepAlive;
             webRequest.UserAgent = UserAgent;
-            webRequest.ConnectionGroupName = ConnectionGroupName;
-            webRequest.ProtocolVersion = ProtocolVersion;
+            webRequest.ConnectionGroupName = ConnectionGroupName ?? webRequest.ConnectionGroupName;
```

Comment:
What is the reason behind the change in these 3 lines? This undoes partially the commit https://github.com/xapi-project/xen-api-sdk/commit/f094386763b41acfb1426eefbcdee551bb151374 (nasty bug...). It's a tricky area that I'd leave untouched unless we have evidence of malfunction.

Reply:
I see.

The issue here is that the values I null coalesced are null.

They were initially instantiated based on the values of XmlRpcProxy, but since that doesn't exist anymore, I just assumed the sensible thing was to use whatever deafult value we had.

now, out of the three values:

Reply:
right so I had a look at the previous behaviour, to check which values we might want to have.

### `ConnectionGroupName`

 This one is actually set to `null` by default, too.
![image](https://user-images.githubusercontent.com/32554698/193239550-ebeb4dd9-fb0a-4722-985c-cead22ff5c26.png).
So no need to coalesce it.

### `ProtocolVersion`

The protocol version defaults to 1.1 when using the XmlRpc version of this code, which is the same one that the new requests are set to using the new logic. In this case, if we don't specify it, we'll get an error thrown at us.

I think we should leave it like this:
```csharp
webRequest.ProtocolVersion = ProtocolVersion ?? webRequest.ProtocolVersion;
```

this is the variable's value when logging in with the old SDK:

![image](https://user-images.githubusercontent.com/32554698/193242103-fdb60a82-d23a-4eeb-a1c1-0bb840110c2c.png)


### `CookieContainer`

 this one is the weirdest one. When using XMLRPC, it has a decent default:
![image](https://user-images.githubusercontent.com/32554698/193242070-70cef4dc-5990-4a9c-8933-69fd181c46e7.png)

but when using the value from webrequest, we're not getting any instantiated cookie container. I checked the XMLRPC library, and it's being created with a normal `new CookieContainer()`

While the value can be null without errors, if we want to replicate the old behaviour, we could do this:
```csharp
 webRequest.CookieContainer = Cookies ?? webRequest.CookieContainer ?? new CookieContainer(); 
 ````
 
 the code above results in the following initial value of the cookie container, which is the same as we previously had. 
 
![image](https://user-images.githubusercontent.com/32554698/193243672-e4f74363-a191-4a45-bceb-833879b4ef93.png)


_______________________________

If this all works for you, I'll push a new commit with the tweaks.